packages/strapi-hook-mongoose/lib/index.js

```diff
@@ -88,16 +88,29 @@ module.exports = function(strapi) {
         connectOptions.useCreateIndex = true;
         connectOptions.useUnifiedTopology = useUnifiedTopology || 'false';
 
-        try {
+        let connect = function() {
           /* FIXME: for now, mongoose doesn't support srv auth except the way including user/pass in URI.
            * https://github.com/Automattic/mongoose/issues/6881 */
-          await instance.connect(
-            uri ||
+          return instance.connect(
+              uri ||
               `mongodb${isSrv ? '+srv' : ''}://${username}:${encodeURIComponent(
-                password
+                  password
               )}@${host}${!isSrv ? ':' + port : ''}/`,
-            connectOptions
+              connectOptions
           );
+        }
+
+        try {
+          await connect();
+
+          instance.connection.on('disconnected', function() {
+            setTimeout(function() {
+              connect()
+              .catch(function(e) {
```

Comment:
You should be handling the connection error

Reply:
I do it with catch clause. 
No need to call Mongoose.connect inside catch. The 'disconnected' event is emitted again.
Catch here is to prevent Unhandled exception log message.

Reply:
I believe he means passing the error through the strapi error handling with a proper message. @AHgPuK

Reply:
You are not handling the error you are just catching it. You should at least be logging the error message.